ballerina/client.bal

```diff
@@ -1,15 +1,3629 @@
+    }
+
+    resource isolated function delete rest/api/'3/project/[string projectIdOrKey](map<string|string[]> headers = {}, *DeleteProjectQueries queries) returns http:Response|error {
+        string resourcePath = string `/rest/api/3/project/${getEncodedUri(projectIdOrKey)}`;
+        resourcePath = resourcePath + check getPathForQueryParam(queries);
+        return self.clientEp->delete(resourcePath, headers = headers);
+    }
+
+    resource isolated function delete rest/api/'3/project/[string projectIdOrKey]/avatar/[int id](map<string|string[]> headers = {}) returns http:Response|error {
+        string resourcePath = string `/rest/api/3/project/${getEncodedUri(projectIdOrKey)}/avatar/${getEncodedUri(id)}`;
+        return self.clientEp->delete(resourcePath, headers = headers);
+    }
+
+    resource isolated function delete rest/api/'3/project/[string projectIdOrKey]/classification\-level/default(map<string|string[]> headers = {}) returns json|error {
+        string resourcePath = string `/rest/api/3/project/${getEncodedUri(projectIdOrKey)}/classification-level/default`;
+        return self.clientEp->delete(resourcePath, headers = headers);
+    }
+
+    resource isolated function delete rest/api/'3/project/[string projectIdOrKey]/properties/[string propertyKey](map<string|string[]> headers = {}) returns http:Response|error {
+        string resourcePath = string `/rest/api/3/project/${getEncodedUri(projectIdOrKey)}/properties/${getEncodedUri(propertyKey)}`;
+        return self.clientEp->delete(resourcePath, headers = headers);
+    }
+
+    resource isolated function delete rest/api/'3/project/[string projectIdOrKey]/role/[int id](map<string|string[]> headers = {}, *DeleteActorQueries queries) returns http:Response|error {
+        string resourcePath = string `/rest/api/3/project/${getEncodedUri(projectIdOrKey)}/role/${getEncodedUri(id)}`;
+        resourcePath = resourcePath + check getPathForQueryParam(queries);
+        return self.clientEp->delete(resourcePath, headers = headers);
+    }
+
+    resource isolated function delete rest/api/'3/projectCategory/[int id](map<string|string[]> headers = {}) returns http:Response|error {
+        string resourcePath = string `/rest/api/3/projectCategory/${getEncodedUri(id)}`;
+        return self.clientEp->delete(resourcePath, headers = headers);
+    }
+
+    resource isolated function delete rest/api/'3/resolution/[string id](map<string|string[]> headers = {}, *DeleteResolutionQueries queries) returns http:Response|error {
+        string resourcePath = string `/rest/api/3/resolution/${getEncodedUri(id)}`;
+        resourcePath = resourcePath + check getPathForQueryParam(queries);
+        return self.clientEp->delete(resourcePath, headers = headers);
+    }
+
+    resource isolated function delete rest/api/'3/role/[int id](map<string|string[]> headers = {}, *DeleteProjectRoleQueries queries) returns http:Response|error {
```

Comment:
I'm seeing many connector operations are returning either json or `http:Response` type instead of returning proper record types. 

@lnash94 Shall we evaluate few such scenarios and and see if that's the expected behaviour or, can we fix/improve our OpenAPI tool tool to generate named records in such scenarios as well?

Reply:
@NipunaRanasinghe For one operation (API),  the types.bal file missed certain attributes that should be generated, so I had to manually add it accordingly. (I do not check it was a mistake on the swagger documentation of jira or your OpenAPI tool missed it while generating. If possible, have a look on those issues too).

Reply:
@NipunaRanasinghe In this case, the given operation has a success status code without a specific body, and in the HTTP client, we cannot directly bind the status code as `http:NoContent`. The `http:Response` is used because the operation has multiple responses.

If the user is interested in binding the status code, they need to add the `—status-code-binding` flag when generating the code.

Since currently we only consider the success payload, We can check the feasibility of improving the tool to generate `error?` for multiple responses. In the meantime, until we provide this improvement, we can modify the OAS by removing specific error responses.

[1] OAS sample which success response has no content type; Here success response is 200 and it only has a description. Remaining of the status codes are error status code. 
```yml
responses:
        '204':
          description: Returned if the request is successful.
        '400':
          description: Returned if the request is invalid or if the replacement project role is not found.
        '401':
          description: Returned if the authentication credentials are incorrect or missing.
        '403':
          description: Returned if the user does not have administrative permissions.
        '404':
          description: Returned if the project role being deleted is not found.
        '409':
          description: Returned if the project role being deleted is in use and a replacement project role is not specified in the request.

```
[2] Modified OAS: Simply removed the remaining status code 401, 404, 403, 409 since we bind and process data with the success payloads.

```yml
responses:
        '204':
          description: Returned if the request 
```
Since this involves many operations to modify, I would suggest keeping `Http:Response`, as it would be fine for such scenarios.

Upon further investigation, I noticed that some operations in the given endpoint `/rest/api/3/field/{fieldKey}/option/{optionId}/issue` only return error responses and no success responses. Therefore, having `Http:Response` is expected for such scenarios.

Reply:
@AnsarMahir Would you mind providing us with the record or object you used for modification? We can check on it.

Reply:
@AnsarMahir, No issue at all, take your time :)